rest/server_context_test.go

```diff
@@ -728,3 +728,67 @@ func TestAdminAuthWithX509(t *testing.T) {
 	_, _, err = checkAdminAuth("", base.TestClusterUsername(), base.TestClusterPassword(), httpClient, managementEndpoints, []string{"cluster!admin"}, nil)
 	assert.NoError(t, err)
 }
+
+// Test GetbucketSpec() ServerTLSSkipVerify behaviour
+func TestTLSSkipVerifyGetBucketSpec(t *testing.T) {
+	errorText := "cannot skip server TLS validation and use CA Cert"
+	testCases := []struct {
+		name                string
+		serverTLSSkipVerify *bool
+		caCert              string
+		expectError         bool
+	}{
+		{
+			name:                "CA Provided, explicitly not skipping TLS validation",
+			serverTLSSkipVerify: base.BoolPtr(false),
+			caCert:              "t.ca",
+			expectError:         false,
+		},
+		{
+			name:        "CA Provided only",
+			caCert:      "t.ca",
+			expectError: false,
+		},
+		{
+			name:                "CA Provided and skipping TLS validation",
+			serverTLSSkipVerify: base.BoolPtr(true),
+			caCert:              "t.ca",
+			expectError:         true,
+		},
+		{
+			name:                "Skipping TLS validation, no CA",
+			serverTLSSkipVerify: base.BoolPtr(true),
+			caCert:              "",
+			expectError:         false,
+		},
+		{
+			name:        "No CA, no TLS validation skip",
+			expectError: false,
+		},
+		{
+			name:                "No CA, no TLS validation skip explicitly",
+			serverTLSSkipVerify: base.BoolPtr(false),
+			expectError:         false,
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			startupConfig := &StartupConfig{Unsupported: UnsupportedConfig{ServerTLSSkipVerify: test.serverTLSSkipVerify}}
+			dbConfig := &DbConfig{BucketConfig: BucketConfig{CACertPath: test.caCert}}
+			spec, err := GetBucketSpec(dbConfig, startupConfig)
+
+			if test.expectError {
+				assert.EqualError(t, err, errorText)
+				assert.Empty(t, spec)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, spec.CACertPath, test.caCert)
```

Comment:
expected comes first

```suggestion
				assert.Equal(t, test.caCert, spec.CACertPath)
```